lib/presentation/features/heroes/bloc/heroes_bloc.dart

```diff
@@ -7,30 +7,59 @@ import 'package:marvel/presentation/convertors/convertor.dart';
 import 'package:marvel/presentation/view_models/view_data_character.dart';
 
 class HeroesBloc extends Bloc<HeroesEvent, HeroesState> {
-  final GetCharactersUseCase getCharacters;
+  final GetCharactersUseCase getCharactersUseCase;
+  final int _offset = 0;
 
-  HeroesBloc({required this.getCharacters})
+  HeroesBloc({required this.getCharactersUseCase})
       : super(
-          const HeroesState(loading: true),
+          const HeroesState(loading: false, hasReachedMax: false),
         ) {
-    on<HeroesEvent>(
+    on<ReadyForDataEvent>(
       (event, emit) async {
         emit(
           state.copyWith(loading: true),
         );
         try {
-          final List<Character> characters = await getCharacters.call();
+          final List<Character> characters = await getCharactersUseCase.call(_offset);
           final List<ViewDataCharacter> charactersViewData = characters
               .map(
                 (characterEntity) => characterEntity.charactersToViewData(characterEntity),
               )
               .toList();
+          emit(
+            state.copyWith(loading: false, characters: charactersViewData),
+          );
+        } catch (e) {
           emit(
             state.copyWith(
               loading: false,
-              characters: charactersViewData,
+              error: "error",
```

Comment:
You show this error-message to the users, right? So it would be good to name it like "We cannot load Heros right now"